flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -395,8 +397,8 @@ private static StateBackend loadChangelogStateBackend(
             Constructor<? extends DelegatingStateBackend> constructor =
                     Class.forName(CHANGELOG_STATE_BACKEND, false, classLoader)
                             .asSubclass(DelegatingStateBackend.class)
-                            .getConstructor(StateBackend.class);
-            return constructor.newInstance(backend);
+                            .getConstructor(StateBackend.class, StateChangelogWriterFactory.class);
+            return constructor.newInstance(backend, new InMemoryStateChangelogWriterFactory());
```

Comment:
I've opened FLINK-21804 to load via plugin manager
and propose to use simple constructor + if/else here for now.

Reply:
Removed this change (as per comment [below](https://github.com/apache/flink/pull/15200#discussion_r640955392))